--- a/engauge.pro
+++ b/engauge.pro
@@ -746,16 +746,11 @@
 pdf {
     message("PDF support:      yes")
     _POPPLER_INCLUDE = $$(POPPLER_INCLUDE)
-    _POPPLER_LIB = $$(POPPLER_LIB)
     isEmpty(_POPPLER_INCLUDE) {
       error("POPPLER_INCLUDE and POPPLER_LIB environment variables must be defined")
-    } else {
-      isEmpty(_POPPLER_LIB) {
-        error("POPPLER_INCLUDE and POPPLER_LIB environment variables must be defined")
-      }
-    }
+		} 
     DEFINES += "ENGAUGE_PDF"
-    LIBS += -L$$(POPPLER_LIB) -lpoppler-qt5
+    LIBS += -lpoppler-qt5
     INCLUDEPATH += $$(POPPLER_INCLUDE)
     HEADERS += src/Dlg/DlgPdfFrame.h \
                src/Pdf/Pdf.h \
