Description: Fix handling of libpoppler in build system 
Last-Update: 2016-07-09

--- a/engauge.pro
+++ b/engauge.pro
@@ -788,16 +788,11 @@
 pdf {
     message("PDF support:      yes")
     _POPPLER_INCLUDE = $$(POPPLER_INCLUDE)
-    _POPPLER_LIB = $$(POPPLER_LIB)
     isEmpty(_POPPLER_INCLUDE) {
       error("POPPLER_INCLUDE and POPPLER_LIB environment variables must be defined")
-    } else {
-      isEmpty(_POPPLER_LIB) {
-        error("POPPLER_INCLUDE and POPPLER_LIB environment variables must be defined")
-      }
-    }
+		} 
     DEFINES += "ENGAUGE_PDF"
-    LIBS += -L$$(POPPLER_LIB) -lpoppler-qt5
+    LIBS += -lpoppler-qt5
     INCLUDEPATH += $$(POPPLER_INCLUDE)
     HEADERS += src/Dlg/DlgImportCroppingPdf.h \
                src/Import/ImportCroppingUtilPdf.h \
