Description: Add hardening flags 
Last-Update: 2015-11-15

--- a/engauge.pro
+++ b/engauge.pro
@@ -591,3 +591,8 @@
       message(Building release version without internal support for JPEG2000 files)
     }
 }
+
+QMAKE_CPPFLAGS *= $(shell dpkg-buildflags --get CPPFLAGS)
+QMAKE_CFLAGS   *= $(shell dpkg-buildflags --get CFLAGS)
+QMAKE_CXXFLAGS *= $(shell dpkg-buildflags --get CXXFLAGS)
+QMAKE_LFLAGS   *= $(shell dpkg-buildflags --get LDFLAGS)
