Description: Add hardening flags 
Last-Update: 2015-11-15

--- a/engauge.pro
+++ b/engauge.pro
@@ -711,3 +711,8 @@
                translations/engauge_pt.ts \
                translations/engauge_ru.ts \
                translations/engauge_zh.ts
+
+QMAKE_CPPFLAGS *= $(shell dpkg-buildflags --get CPPFLAGS)
+QMAKE_CFLAGS   *= $(shell dpkg-buildflags --get CFLAGS)
+QMAKE_CXXFLAGS *= $(shell dpkg-buildflags --get CXXFLAGS)
+QMAKE_LFLAGS   *= $(shell dpkg-buildflags --get LDFLAGS)
