Description: Add doc directory to search path for help 
Last-Update: 2015-10-25

--- a/src/Help/HelpWindow.cpp
+++ b/src/Help/HelpWindow.cpp
@@ -53,6 +53,7 @@
 #endif
   paths << "/documentation/engauge.qhc";
   paths << "/../share/doc/engauge-digitizer/engauge.qhc";
+  paths << "/usr/share/engauge-digitizer-doc/engauge.qhc";
 
   QStringList::iterator itr;
   for (itr = paths.begin(); itr != paths.end(); itr++) {
