Description: Add doc directory to search path for help 
Last-Update: 2015-10-25

Index: engauge-digitizer/src/Help/HelpWindow.cpp
===================================================================
--- engauge-digitizer.orig/src/Help/HelpWindow.cpp
+++ engauge-digitizer/src/Help/HelpWindow.cpp
@@ -67,6 +67,7 @@ QString HelpWindow::helpPath() const
 #else
   paths << "/documentation/engauge.qhc";
   paths << "/../share/doc/engauge-digitizer/engauge.qhc";
+  paths << "/usr/share/engauge-digitizer-doc/engauge.qhc";
 #endif
 
   QStringList::iterator itr;
