Description: Fix target location for makefile
Last-Update: 2015-10-25

Index: engauge-digitizer/src/engauge.pro
===================================================================
--- engauge-digitizer.orig/src/engauge.pro
+++ engauge-digitizer/src/engauge.pro
@@ -489,7 +489,7 @@ SOURCES += \
     View/ViewSegmentFilter.cpp \
     util/Xml.cpp
 
-TARGET = ../bin/engauge
+TARGET = ./bin/engauge
 
 QT += core gui network printsupport widgets xml help
 
